site/docs/v1/tech/admin-guide/configuration-management/terraform.adoc

```diff
@@ -0,0 +1,502 @@
+
+[source]
+----
+resource "fusionauth_user" "forum-user1" {
+  email                    = "forum-user1@email.internal"
+  first_name               = "John"
+  last_name                = "Doe"
+  middle_name              = "William"
+  password_change_required = true
+  password                 = "%WLTvrsYELsyPqC^R7FMUNxt##VyDf6XaWk2R7!gS$oL76Ww"
+  username_status          = "ACTIVE"
+  lifecycle {
+    ignore_changes = [
+      data,
+      image_url,
+      password
+    ]
+  }
+}
+
+resource "fusionauth_user" "forum-admin1" {
+  email                    = "forum-admin1@email.internal"
+  first_name               = "John"
+  last_name                = "Doe"
+  middle_name              = "William"
+  password_change_required = true
+  password                 = "@CfosPAVT3&hCzz5c^&#2F5BxNUY$X!@s!7Wx9bd6Yon54e3"
+  username_status          = "ACTIVE"
+  lifecycle {
+    ignore_changes = [
+      data,
+      image_url,
+      password
+    ]
+  }
+}
+----
+
+Another example is where your e.g. business application adds user driven data to the FusionAuth Application.
+
+[source]
+----
+resource "fusionauth_application" "forum" {
+  tenant_id = data.fusionauth_tenant.Default.id
+  name = "forum"
+  lifecycle {
+    ignore_changes = [
+      data
+    ]
+  }
+}
+----
+
+If you are only interested in initially creating and destroying resources you can as well ignore all changes.
+
+[source]
+----
+lifecycle {
+  ignore_changes = all
+}
+----
+
+== Terraform Remove
+
+If you want to remove a resource you can comment or delete the lines accordingly. Run `terraform plan` and `terraform apply` once modified which will then destroy the commented or deleted ressource.
+
+If you run link:https://developer.hashicorp.com/terraform/cli/commands/destroy[terraform destroy] you need to be aware that all Terraform managed resources will be destroyed.
+
+=== Prevent Destroy
+If you still want to make sure that certain resources arent destroyed, you can specify the link:https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#prevent_destroy[prevent_destroy] lifecycle meta-argument as a measure of safety against the accidental replacement of objects that may be costly to reproduce. However, it will make certain configuration changes impossible to apply, and will prevent the use of the `terraform destroy` command once such objects are created, and so this option should be used sparingly.
+
+[source]
+----
+lifecycle {
+  prevent_destroy = true
+}
+----
+
+A good example would be a productive FusionAuth tenant, as deleting a tenant removes all of the associated applications, groups and users.
+
+The same meta-argument you can use for Terraform managed non-deletable resources like Default tenant and FusionAuth application as Terraform would not be able to anyway.
```

Comment:
Should we introduce this earlier like when you are importing the fusionauth application and default tenant?

Reply:
Yes that's a good idea, I will reference this part earlier but I think it's important that the full Prevent Destroy section is in the Terraform Remove section.

Reply:
Makes sense to repeat it here.